Core/Object Arts/Dolphin/Base/BlockClosure.cls

```diff
@@ -549,6 +549,33 @@ stackTempCount: anInteger
 	info := (info bitAnd: ##((16rFF << 15) bitInvert)) 
 				bitOr: ((anInteger bitAnd: 16rFF) bitShift: 15)!
 
+storeOn: aStream
+	"Append, to aStream, a <String> whose characters can recreate the receiver."
+
+	| m block class selector string |
+	m := self method.
+	(m notNil and: [m isExpression]) ifTrue: [	"We can print the source"
```

Comment:
You are right. Sometimes there is no source so you need a decompiler or
keep the block as literal.

You should be aware that the VC allows users to touch trough the inspector
other values than those that can be edited via aspect editors.

Maybe you will encounter views larger than the maximum input string size
for the compiler.

Best Regards
El mar. 15, 2016 9:08 PM, "James Foster" <notifications@github.com>
escribió:

> In Core/Object Arts/Dolphin/Base/BlockClosure.cls
> <https://github.com/dolphinsmalltalk/Dolphin/pull/160#discussion_r56263218>
> :
>
> > @@ -549,6 +549,33 @@ stackTempCount: anInteger
> >  	info := (info bitAnd: ##((16rFF << 15) bitInvert))
> >  				bitOr: ((anInteger bitAnd: 16rFF) bitShift: 15)!
> >
> > +storeOn: aStream
> > +	"Append, to aStream, a <String> whose characters can recreate the receiver."
> > +
> > +	| m block class selector string |
> > +	m := self method.
> > +	(m notNil and: [m isExpression]) ifTrue: [	"We can print the source"
>
> In existing views, the literal array creates a BlockClosure that names a
> source method but may be completely different so there isn't really any
> source anywhere.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/dolphinsmalltalk/Dolphin/pull/160/files/d43e0d4b6526ded0cca71e5aa0b4d63e0330ee55#r56263218>
>